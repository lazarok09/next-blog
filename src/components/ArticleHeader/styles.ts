import styled, { css } from "styled-components";
import { Title as HeadingStyles } from "../Heading/styles";

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      @media ${theme.media.lteMedium} {
        font-size: calc(${theme.font.sizes.medium} + 1rem);
      }
    }
  `}
`;
export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
    @media ${theme.media.lteMedium} {
      font-size: calc(${theme.font.sizes.small} + 0.2rem);
    }
  `}
`;
export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
