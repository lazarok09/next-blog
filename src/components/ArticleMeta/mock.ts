import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: 'aowea3i902103102',

    slug: 'lazaro-souza',
    displayName: 'LÃ¡zaro Souza',
  },
  categories: [
    {
      id: '123125129120',
      displayName: 'JS',
      slug: 'javascript',
    },
    {
      id: '123125129120125512',
      displayName: 'Tech',
      slug: 'tech',
    },
  ],
} as ArticleMetaProps;
