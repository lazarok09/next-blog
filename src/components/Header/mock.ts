import { HeaderProps } from '.';

export default {
  blogName: 'LÃ¡zaro',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: 'https://github.com/lazarok09.png',
  showText: true,
} as HeaderProps;
