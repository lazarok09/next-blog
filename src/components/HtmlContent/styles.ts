import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    p {
      margin: ${theme.spacings.xlarge} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      transition: all 300ms ease-in-out;
      text-decoration: none;
    }
    a:hover {
      filter: brightness(80%);
      text-decoration: underline;
    }
    pre, code{
      background: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.xlarge} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
  }
    img {
        width: 100%;

    }
    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.xlarge} 0;
      max-width: 100%;

    }
    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }
    .image-style-side {
      float: right;
      max-width: 50%;
      margin: 0 ${theme.spacings.medium};
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }
    ul,
    ol {
      margin: ${theme.spacings.xlarge};
    }
    .table {
      max-width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 100%;
      border-collapse: collapse;
    }
    table th,
    table td{
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }
    }
    iframe {
    max-width: 100%;
    height: 50vh;
    }
    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: 0;
      }
    }
  `}
`;
