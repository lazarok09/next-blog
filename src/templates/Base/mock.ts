import { BaseTemplateProps } from '.';
import { data } from '../../api/data.json';
export default {
  settings: data.setting,
  children: 'Hello World',
} as BaseTemplateProps;
