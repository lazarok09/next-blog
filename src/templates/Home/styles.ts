import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    & > img {
      height: 40px;
    }
    background: ${theme.colors.white};
  `}
`;
