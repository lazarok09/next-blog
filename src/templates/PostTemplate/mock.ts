import { PostTemplateProps } from '.';
import { data } from '../../api/data.json';
export default {
  settings: data.setting,
  post: data.posts[0],
} as PostTemplateProps;
