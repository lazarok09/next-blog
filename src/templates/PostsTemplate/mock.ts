import { data } from '../../api/data.json';
import { PostsTemplateProps } from '.';

export default {
  settings: data.setting,
  posts: data.posts,
} as PostsTemplateProps;
