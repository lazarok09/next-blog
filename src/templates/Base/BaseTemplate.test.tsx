import { screen } from '@testing-library/react';
import { BaseTemplate, BaseTemplateProps } from '.';
import { renderTheme } from '../../styles/render-theme';
import mock from './mock';

const props: BaseTemplateProps = mock;
describe('<BaseTemplate />', () => {
  it('should render a image heading with alt text', () => {
    renderTheme(<BaseTemplate {...props} />);
    expect(
      screen.getByRole('img', {
        name: 'challenge accepted - A blog to talk about programing and tecnology',
      }),
    ).toBeInTheDocument();
  });
  it('should render a menu and his icon', () => {
    renderTheme(<BaseTemplate {...props} />);
    expect(screen.getByLabelText('Open or close menu')).toBeInTheDocument();
  });
  it('should render a footer with text', () => {
    renderTheme(<BaseTemplate {...props} />);
    expect(screen.getByText('texto')).toBeInTheDocument();
  });
  it('should render a go to top component', () => {
    renderTheme(<BaseTemplate {...props} />);
    expect(screen.getByLabelText(/Go to Top/i)).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<BaseTemplate {...props} />);
    expect(container).toMatchSnapshot();
  });
});
