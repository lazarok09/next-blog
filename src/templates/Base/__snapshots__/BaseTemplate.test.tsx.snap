// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseTemplate /> should match snapshot 1`] = `
.c15 {
  text-align: center;
  font-size: 1.6rem;
  border-top: 0.1rem solid #DDDDDD;
}

.c15 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c16 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c16 p {
  margin: 2.4rem 0;
}

.c16 a,
.c16 a:visited,
.c16 a:link {
  color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c16 a:hover {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c16 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c16 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 4.0rem 0;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c16 pre code {
  color: inherit;
  background: inherit;
}

.c16 img {
  width: 100%;
}

.c16 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
  max-width: 100%;
}

.c16 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c16 .image-style-side {
  float: right;
  max-width: 50%;
  margin: 0 2.4rem;
}

.c16 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c16 ul,
.c16 ol {
  margin: 2.4rem 4.0rem;
}

.c16 .table {
  max-width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c16 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c16 table th,
.c16 table td {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c16 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c16 iframe {
  max-width: 100%;
  height: 50vh;
}

.c17 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 6;
  bottom: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  opacity: 0.8;
  color: #FFFFFF;
  background: #000000;
}

.c17:hover {
  opacity: 1;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c12 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  height: 15rem;
  width: 15rem;
  border-radius: 50%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c10 .c5 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c8 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c2 {
  background: #000000;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: -31rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c4 .c5 img {
  border: 0.5rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #000000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: 1rem;
  color: #FFFFFF;
}

.c9 {
  margin-top: 6.4rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c14 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c16 {
    font-size: 2rem;
  }

  .c16 .image-style-side {
    float: none;
    max-width: 100%;
    margin: 0;
  }
}

@media (max-width:768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width:768px) {
  .c0 {
    left: -0.5rem;
  }
}

<div>
  <div
    class=""
  >
    <a
      aria-label="Open or close menu"
      class="c0"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              class="c7"
              href="/"
              target="_self"
            >
              <img
                alt="challenge accepted"
                src="https://res.cloudinary.com/lazarok09/image/upload/v1633795031/icon_4bc14e0cf4.png"
              />
            </a>
          </h1>
        </div>
        <a
          class="c8"
          href="/"
          target="_self"
        >
          home
        </a>
      </nav>
    </div>
    <div
      class="c9"
    >
      <header
        class="c10"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="challenge accepted - A blog to talk about programing and tecnology"
              src="https://res.cloudinary.com/lazarok09/image/upload/v1633795031/icon_4bc14e0cf4.png"
            />
          </a>
        </h1>
        <div
          class="c11"
        >
          <h2
            class="c5 c12"
          >
            challenge accepted
          </h2>
          <p>
            A blog to talk about programing and tecnology
          </p>
        </div>
      </header>
    </div>
    <div
      class="c13"
    >
      Hello World
    </div>
    <div
      class="c14"
    >
      <footer
        class="c15"
      >
        <div
          class="c16"
        >
          <p>
            texto
          </p>
        </div>
      </footer>
    </div>
    <a
      aria-label="Go to Top"
      class="c17"
      href="#"
      title="Go to Top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
